### PR TITLE
Update to DEVELOPMENT-SNAPSHOT-2016-05-03.

### PR DESCRIPTION
**Pull request by mjmsmith**
Changes required to compile on the 05/03 snapshot. Relevant proposals:

[SE-0049: Move @noescape and @autoclosure to be type attributes](https://github.com/apple/swift-evolution/blob/master/proposals/0049-noescape-autoclosure-type-attrs.md)
[SE-0065: A New Model For Collections and Indices](https://github.com/apple/swift-evolution/blob/master/proposals/0065-collections-move-indices.md)

### CODE REVIEW

**Comment by tanner0101** [comment]
@mjmsmith You are a god. I started working on this two nights ago but got overwhelmed. 🙏

**Comment by tanner0101** [comment]
We should add at least one unit test to make sure this still works. Something as simple as:

``` swift
["value": "42"]
```

``` mustache
Test {{ value }} 
```

``` swift
XCTAssert(rendered == "Test 42")
```

**Comment by antonmes** [comment]
awesome!

**Comment by paulofaria** [comment]
:scream: haha @tannernelson the same happened to me hahahaj

**Comment by mjmsmith** [comment]
Duh, I only just realized you can run `swift test` without needing to set anything else up.  Will add a test or two.

**Comment by tanner0101** [comment]
@mjmsmith You can also run the tests in Xcode. Do `swift build -X` to generate the project, then hit `Command+U`. 

Just make sure to select the correct toolchain (Xcode > Toolchain)

File: Sources/Mustache/Parsing/TemplateParser.swift [hunk 1]
```diff
@@ -362,19 +364,19 @@ final class TemplateParser {
...
+          tagStartLength = tagDelimiterPair.0.distance(from: tagDelimiterPair.0.startIndex, to: tagDelimiterPair.0.endIndex)
+          tagEndLength = tagDelimiterPair.1.distance(from: tagDelimiterPair.1.startIndex, to: tagDelimiterPair.1.endIndex)
 
             let usesStandardDelimiters = (tagDelimiterPair.0 == "{{") && (tagDelimiterPair.1 == "}}")
             unescapedTagStart = usesStandardDelimiters ? "{{{" : nil
-            unescapedTagStartLength = unescapedTagStart != nil ? unescapedTagStart!.startIndex.distance(to: unescapedTagStart!.endIndex) : 0
+          unescapedTagStartLength = unescapedTagStart != nil ? unescapedTagStart!.distance(from: unescapedTagStart!.startIndex, to: unescapedTagStart!.endIndex) : 0
             unescapedTagEnd = usesStandardDelimiters ? "}}}" : nil
-            unescapedTagEndLength = unescapedTagEnd != nil ? unescapedTagEnd!.startIndex.distance(to: unescapedTagEnd!.endIndex) : 0
+          unescapedTagEndLength = unescapedTagEnd != nil ? unescapedTagEnd!.distance(from: unescapedTagEnd!.startIndex, to: unescapedTagEnd!.endIndex) : 0
```
Comment (tanner0101) [inline Sources/Mustache/Parsing/TemplateParser.swift:374]: I'd vote for removing the line length limit.

**Comment by mjmsmith** [comment]
Tests added, fwiw also worked fine with my own project's [templates](https://github.com/mjmsmith/starredsearch/tree/master/Resources/Views).

**Comment by antonmes** [comment]
@paulofaria you missed? you promised shoot this dog. 😂

**Comment by tanner0101** [comment]
Tests look great. Just need to update the `.swift-version` file so that Travis will work.

**Comment by houndci-bot-dead** [comment]
they told me one day that i may add as many annoying notes as i want to, but look at me now..
so please, don't.

**Comment by paulofaria** [comment]
@houndci-bot-dead hahahaha can't believe you did this @antonmes  hahahahahahahah

**Comment by antonmes** [comment]
me? 😱

**Comment by paulofaria** [comment]
LGTM: @mjmsmith can I merge?

**Comment by paulofaria** [comment]
@mjmsmith are you at [slack.zewo.io](http://slack.zewo.io)?

**Pull request merged**